tests/unit/test_paging.py

```diff
@@ -0,0 +1,98 @@
+import requests
+import json
+
+from tests.framework import (CapturedIOTestCase)
+from globus_sdk.transfer.paging import PaginatedResource
+from globus_sdk.transfer.response import IterableTransferResponse
+
+
+class PagingSimulator():
+
+    def __init__(self, n):
+        self.n = n  # the number of simulated items
+
+    def simulate_get(self, path, params=None,
+                     headers=None, response_class=None, retry_401=True):
+        """
+        Simulates a paginated response from a Globus API get supporting limit,
+        offset, and has next page
+        """
+        offset = params["offset"]
+        limit = params["limit"]
+        data = {}  # dict that will be treated as the json data of a response
+        data["offset"] = offset
+        data["limit"] = limit
+        # fill data field
+        data["DATA"] = []
+        for i in range(offset, min(self.n, offset + limit)):
+            data["DATA"].append({"value": i})
+        # fill has_next_page field
+        data["has_next_page"] = (offset + limit) < self.n
+
+        # make the simulated response
+        response = requests.Response()
+        response._content = bytes(json.dumps(data))
+        response.headers["Content-Type"] = "text/plain"
```

Comment:
I assume something blew up with `Content-Type` unset? If we're going to set it, we should use `application/json`, as that is the one the service will send.